vision_agent_tools/models/siglip.py

```diff
@@ -0,0 +1,90 @@
+from typing import List, Dict, Any
+import torch
+from PIL import Image
+from pydantic import ConfigDict, validate_arguments
+from transformers import AutoProcessor, AutoModel
+from transformers import SiglipProcessor, SiglipModel
+
+from vision_agent_tools.shared_types import BaseMLModel, Device
+
+
+class Siglip(BaseMLModel):
+    """
+    Tool for object detection using the pre-trained Siglip model.
+    This tool takes a prompt as input and generates an image using the Siglip model.
+    """
+
+    config = ConfigDict(arbitrary_types_allowed=True)
```

Comment:
why do you need this?

Reply:
Dont need it! Removing, thanks!

Reply:
I just noticed I really need it because of the `@validade_arguments`
Otherwise we get this error as it validates the PIL Image input:
![image](https://github.com/user-attachments/assets/be96bb0a-1181-493b-aaa6-33d7fcd92d45)